src/components/DropdownFilterContainer.ts

```diff
@@ -0,0 +1,150 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+import * as classNames from "classnames";
+import * as dijitRegistry from "dijit/registry";
+import * as dojoConnect from "dojo/_base/connect";
+import * as dojoLang from "dojo/_base/lang";
+
+import { Alert } from "./Alert";
+import { DropdownFilter, DropdownFilterProps } from "./DropdownFilter";
+import { Utils, parseStyle } from "../utils/ContainerUtils";
+
+interface WrapperProps {
+    class: string;
+    style: string;
+    friendlyId: string;
+    mxform?: mxui.lib.form._FormBase;
+}
+
+export interface ContainerProps extends WrapperProps {
+    defaultFilter: number;
+    entity: string;
+    targetListViewName: string;
+    filters: FilterProps[];
+}
+
+export interface FilterProps {
+    caption: string;
+    filterBy: filterOptions;
+    attribute: string;
+    attributeValue: string;
+    constraint: string;
+}
+
+export type filterOptions = "attribute" | "XPath";
+type HybridConstraint = Array<{ attribute: string; operator: string; value: string; path?: string; }>;
+
+export interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: HybridConstraint | string;
+    };
+    filter: {
+        [key: string ]: HybridConstraint | string;
+    };
+    update: () => void;
+    _entity: string;
+}
+
+export interface DropdownFilterState {
+    widgetAvailable: boolean;
+    targetListView?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+    value?: string;
+}
+
+export default class DropdownFilterContainer extends Component<ContainerProps, DropdownFilterState> {
+
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.state = { widgetAvailable: true };
+        this.handleChange = this.handleChange.bind(this);
+        // Ensures that the listView is connected so the widget doesn't break in mobile due to unpredictable render time
+        dojoConnect.connect(props.mxform, "onNavigation", this, dojoLang.hitch(this, this.connectToListView));
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-dropdown-filter")
+            },
+            this.renderAlert(),
+            this.renderDropdownFilter()
```

Comment:
Render only drop down, not additional div.
Only need to add div is case of alert?
In web modeler preview only add the additional div if there is an alert too try to render as good as it gets.